wordpress-mobile/WordPress-iOS
# Copy comment link from notifications, reader and comment details (my site)

- Added copy text functionality to notifications comments, Reader and comments on my site.
- Updated time stamp labels in the cells presented in the above mentioned view controller to be a new 
   UILabel subclass: LongPressGestureLabel.
  This creates a UILabel with a long press gesture recognizer that executes an action defined by the view 
  controllers when the cells are being configures.
- UIAlertViewController has a helper method to create the action sheet for displaying the options to 
   either copy the text or cancel the action in case the long press was unintentional.

For further discussion: https://mobilerequests.wordpress.com/2018/09/10/get-link-to-a-comment-in-the-mobile-apps/

** Also included a code comment formatting that was suggested in a previous PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/10160

Fixes #10161 

To test:
*Notifications Detail*
1. Go to Notifications tab.
2. Tap comment in notifications list.
3. Long-press timestamp label for comment.
4. Notice message that comment URL is copied.

*Comments Detail*
1. Go to My Site tab.
2. Tap Comments item under Publish section.
3. Tap comment in comments list.
4. Long-press timestamp label on comment.
5. Notice message that comment URL is copied.

*Reader*
1. Go to Reader tab.
2. Tap Comments icon in post footer.
3. Long-press timestamp label for comment in comments list.
4. Notice message that comment URL is copied.

![copy_link](https://user-images.githubusercontent.com/1335657/45993402-aea21600-c043-11e8-893a-a8b85b268ce2.gif)
